Add uno example for printing two lines

vcrn opened the pull request:
This is another example running on Arduino Uno. Since all of the `main.rs` files of the Nano examples seem to run on the Uno without any modifications, it felt a bit redundant to copy-paste one of the examples already available. This example shows how you can write text on both lines of an LCD as well as how you can implement your own method on the `struct LcdDisplay`.

I also updated the `examples/README` to inform about some users experiencing issues with initializing their LCD using the Uno, linking to the open issue.

What do you think?

mjhouse commented on examples/README.md:
```diff
@@ -13,20 +13,20 @@ uncomment lines or otherwise modify these examples to get them to work.
+| d2      | D7       | YES      |
 
-Note, that in the case of Arduino Uno, only setups using required pins have been tested to work.
+Note, regarding Arduino Uno: only setups using required pins have been tested to work; and users have also reported sometimes have issues with initializing the LCD, which is discussed in [this issue](https://github.com/mjhouse/ag-lcd/issues/4).
```
drop the `and` at the beginning, and shorten this sentence. Something like: `Some users have reported issues with initializing the LCD (see [#4](https://github.com/mjhouse/ag-lcd/issues/4))`

mjhouse commented on examples/uno/two_lines/Cargo.toml:
```diff
@@ -0,0 +1,34 @@
+[dependencies]
+panic-halt = "0.2.0"
+embedded-hal = "0.2.3"
+ag-lcd = "0.1.2"
```
I'm not certain, but I think we should build with a relative path to the root directory (`ag-lcd = { path = "../../" }`) rather than with the crates.io package name- the examples exist in part to act as integration tests, although they're not run as part of a CI/CD pipeline, the idea is that if you can build and run the examples from the master branch, then you've just validated the library to some degree.

vcrn replied:
Yeah, I agree. I'll get it changed in uno/print too.

mjhouse commented on examples/uno/two_lines/src/main.rs:
```diff
@@ -0,0 +1,52 @@
+    fn print_two_lines(&mut self, _first_row: &str, _second_row: &str) {}
+}
+
+impl<T, D> PrintTwoLines for LcdDisplay<T, D>
```
I think maybe the custom trait example could be in it's own project- I'd prefer to have each example demonstrate one, clearly defined thing. If possible- obviously print demos are going to include setup, clear etc as well.

vcrn replied:
I'll avoid defining a function also for it in that case, something like this:
`        
        self.set_position(0, 0);
        self.print("Hello");
        arduino_hal::delay_us(100); 
        self.set_position(0, 1);
        self.print("World");
`

mjhouse replied:
Actually, nevermind. I'm a little slow today- this example is to demonstrate the custom trait stuff, the fact that you're doing it by making a two-line print function is beside the point. Ignore this comment. Sorry.

vcrn commented:
I probably won't be able to contribute any code during the coming month or so, but I'll get your comments resolved as soon as I'm back. 

I'll be available via email and github though, if you want to discuss something.

vcrn commented:
I was able to keep working on this PR much sooner than I expected. I've made your suggested changes. and also updated the relative paths of the nano/examples to ag-lcd since they were no longer working because of the new subdirectory `nano/`. I also updated the license of the Uno-examples so they match that of the crate.

What do you think? And if you've changed your mind regarding the `impl PrintTwoLines` and want to go with what you first suggested, that's fine with me.

mjhouse reviewed (approved):
Looks good. Sorry it took me so long to get around to this.

vcrn commented:
No problem at all! Less than 24 hours is nothing to complain about
